### DISCUSSION TITLE
Credentials callback returning 401 works locally but not production

### DISCUSSION THREAD

**Discussion by MichaelTao** [nextauthjs/next-auth#6789] [Help]
I've been using NextAuth on my site for a while now and it has been working properly. One day it stopped and I was getting emails from clients who can't log in. 

Everything works fine locally, but on production just refuses to.

I spent the whole week trying to figure this out and I'm just lost at this point on what is going on.

Variables all set properly

NEXTAUTH_URL=https://domain.com not localhost:3000 etc..
NEXTAUTH_SECRET
API_LOGIN_URL

**Things I've tried:**

- Updated **NEXTAUTH_URL** tried https://domain.com and even vercel.app domain they assign.
- Spun up a new NextJS 13 with only next auth to rule out other code.
- Deployed it on Vercel and Netlify
- Turned off CORS on the backend server so nothing is blocking it.
- Simplified the [...nextauth].js file so it's only returning the JWT token

**package.json**

![image](https://user-images.githubusercontent.com/1996108/220782490-f474b43e-eaa9-4a1f-8d36-3e391e089fc0.png)

**_app.js**

![image](https://user-images.githubusercontent.com/1996108/220782566-3cc6e716-43d6-4957-8a59-461ac02e6176.png)

**[...nextauth].js**

![image](https://user-images.githubusercontent.com/1996108/220782769-e77e124a-208d-468c-8e31-f3c6011e92a5.png)

**login.js**

![image](https://user-images.githubusercontent.com/1996108/220783026-48f94e7b-49b6-49f1-9f84-b649f4459f87.png)

**returned json when you successfully login**
```
{
  "jwt": "jwt token",
  "user": {
    "id": 16,
    "username": "username",
    "email": "example@domain.com",
    "provider": "local",
    "confirmed": true,
    "blocked": false,
    "employer": true,
    "trial": false,
    "trialExpireAt": null,
    "createdAt": "2023-02-18T00:30:18.977Z",
    "updatedAt": "2023-02-18T00:30:33.745Z",
    "first_name": null,
    "last_name": null
  }
}
```
![image](https://user-images.githubusercontent.com/1996108/220784205-7987cf65-8ef5-4cf9-b8d6-273242e8fd30.png)

**Comment by atridadl** [comment]
Same issue here, but using Github OAuth + SvelteKit... O_O
